autopilot/prefattach_test.go

```diff
@@ -73,29 +73,23 @@ func TestPrefAttachmentSelectEmptyGraph(t *testing.T) {
 		NewNodeID(pub): {},
 	}
 
-	for _, graph := range chanGraphs {
-		success := t.Run(graph.name, func(t1 *testing.T) {
-			graph, err := graph.genFunc(t1)
-			if err != nil {
-				t1.Fatalf("unable to create graph: %v", err)
-			}
+	for _, chanGraph := range chanGraphs {
+		chanGraph := chanGraph
+		graph, err := chanGraph.genFunc(t)
+		require.NoError(t, err)
 
+		success := t.Run(chanGraph.name, func(t1 *testing.T) {
 			// With the necessary state initialized, we'll now
 			// attempt to get the score for this one node.
 			const walletFunds = btcutil.SatoshiPerBitcoin
-			scores, err := prefAttach.NodeScores(graph, nil,
-				walletFunds, nodes)
-			if err != nil {
-				t1.Fatalf("unable to select attachment "+
-					"directives: %v", err)
-			}
+			scores, err := prefAttach.NodeScores(
+				graph, nil, walletFunds, nodes,
+			)
+			require.NoError(t1, err)
```

Comment:
does it make sense to use `NoErrorf` and preserve the original messaging on test failures?

Reply:
The require library creates a nice diff of the expected/actual value and the location where the expectation can be found in the code (including a call stack). And since we probably have to take a look at the code to find out what went wrong anyway, those additional contexts rarely help a lot and make the code harder to read (IMO).
So I added it where it made sense (boolean values) but left the rest as they are.